packages/core/strapi/lib/types/core/attributes/common.d.ts

```diff
@@ -71,4 +71,14 @@ export type Any =
   | Attribute.Text
   | Attribute.Time
   | Attribute.Timestamp
-  | Attribute.UID<Common.UID.Schema | undefined>;
+  | Attribute.UID<Common.UID.Schema>;
+
+export type PopulatableKind = Extract<
+  Attribute.Kind,
+  'relation' | 'component' | 'dynamiczone' | 'media'
+>;
+
+export type NonPopulatableKind = Exclude<
+  Attribute.Kind,
+  'relation' | 'component' | 'dynamiczone' | 'media'
+>;
```

Comment:
Those 2 types have been brought from the entity-service types branch and can safely be ignored. I've let them in the PR since they'll be added soon anyway. We can remove them if someone really feels uncomfortable adding them before their usage :shrug: